src/utils.ts

```diff
@@ -452,13 +467,130 @@ export class Utils {
         }
         return;
     }
+
+    /**
+     * Generates GitHub workflow Summary
+     * This function runs as part of post-workflow cleanup function,
+     * collects existing section markdown files generated by the CLI,
+     * and constructs a single markdown file, to be displayed in the GitHub UI.
+     */
+    public static async generateWorkflowSummaryMarkdown() {
+        // This is the file path that the markdown should be copied to.
+        // In order for GitHub to display this as a job summary in the UI.
+        const githubStepSummaryFilePath: string | undefined = process.env.GITHUB_STEP_SUMMARY;
+        if (!githubStepSummaryFilePath) {
+            core.debug('GITHUB_STEP_SUMMARY is not set. Workflow summary will not be generated.');
+            return;
+        }
+
+        try {
+            // Read all sections and construct the final markdown file
+            const markdownContent: string = await this.readCLIMarkdownSections();
+            if (markdownContent.length == 0) {
+                core.debug('No job summaries sections found. Workflow summary will not be generated.');
+                return;
+            }
+            await fs.writeFile(githubStepSummaryFilePath, markdownContent);
```

Comment:
Can we use the core library instead?
https://github.com/actions/toolkit/blob/main/packages/core/README.md#populating-job-summary
```suggestion
            core.summary.addRaw(markdownContent);
            await core.summary.write();
```